### PR TITLE
DHFPROD-3922: Enable step running in Bench/Flows view

### PR DESCRIPTION
**Pull request by wooldridge**
Includes confirmation dialogs and status spinner.

Screencast: https://drive.google.com/file/d/1ixG2zw1R_FO3eSa8o-owNRlvWx2wjR5o/view

This PR replaces: https://github.com/marklogic/marklogic-data-hub/pull/3652

### CODE REVIEW

**Comment by bsrikan** [comment]
JSON and XML loads are working as expected. CSV is throwing 500 errors. Updated comment in JIRA.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/job/JobDocManager.java [hunk 1]
```diff
@@ -99,6 +100,20 @@ public JsonNode getJobDocumentsForFlows(List<String> flowNames) {
         return getJobDocuments(params);
     }
 
+    public JsonNode getLatestJobDocumentForFlow(String flowName) {
+        RequestParameters params = new RequestParameters();
```
[the remark this answers is not in the document]
Comment (rjrudin) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/job/JobDocManager.java]: Agree - only the controllers are invoking these methods, and there aren't any tests for these methods or for the controller methods (we don't need the latter, do need the former). 

We just had a bug opened up in github for an issue with job counts. I think there's a significant lack of automated tests for everything that is written to the jobs database, which is why we have those bugs pop up.

File: one-ui/src/main/java/com/marklogic/hub/curation/controllers/JobsController.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.Collections;
+import java.util.List;
+
+@Controller
+@RequestMapping(value="/api/jobs")
```
[the remark this answers is not in the document]
Comment (rjrudin) [reply one-ui/src/main/java/com/marklogic/hub/curation/controllers/JobsController.java]: I don't think tests are needed when the controller method is just passing data through, but once we have any logic - like in getJobs and flattenJobsJson - tests are worthwhile.

**Comment by ryanjdew** [comment]
@bsrikan addressed the issue with the separator for text delimited ingest not being populated with a test for that scenario. I also fixed an issue where failed jobs would not record if the error was in the Java-layer with additional tests on the JobDocManager that was missing before.

**Comment by bsrikan** [comment]
This PR has tests for backend. UI has one test that verifies availability of form elements for flows. So to me it doesnt meet the Definition of Done for the story that we agreed upon. There should be one test atleast in flows.test.tsx that tries to mock the run flow API and ensure the mocked instance was called once.

**Comment by wooldridge** [comment]
@bsrikan, I honestly don't have the bandwidth to complete the testing that you're asking for above in this sprint.

**Comment by wooldridge** [comment]
@SameeraPriyathamTadikonda, I'm seeing a build failure, looks like related to the Node.js version update:
```
[2020-03-11T19:24:51.401Z] > Task :one-ui:processResources

[2020-03-11T19:24:51.401Z] > Task :one-ui:classes

[2020-03-11T19:24:51.668Z] > Task :one-ui:bootWar

[2020-03-11T19:24:51.937Z] > Task :one-ui:bootDistTar

[2020-03-11T19:24:53.838Z] > Task :one-ui:bootDistZip

[2020-03-11T19:24:53.838Z] > Task :one-ui:jar SKIPPED

[2020-03-11T19:24:53.838Z] > Task :one-ui:startScripts

[2020-03-11T19:24:53.838Z] > Task :one-ui:distTar

[2020-03-11T19:24:55.748Z] > Task :one-ui:distZip

[2020-03-11T19:24:55.748Z] > Task :one-ui:war SKIPPED

[2020-03-11T19:24:55.748Z] > Task :one-ui:assemble

[2020-03-11T19:24:55.748Z] > Task :one-ui:nodeSetup FAILED

[2020-03-11T19:24:55.748Z] 

[2020-03-11T19:24:55.748Z] FAILURE: Build failed with an exception.

[2020-03-11T19:24:55.748Z] 

[2020-03-11T19:24:55.748Z] * What went wrong:

[2020-03-11T19:24:55.748Z] Execution failed for task ':one-ui:nodeSetup'.

[2020-03-11T19:24:55.748Z] > Could not resolve all files for configuration ':one-ui:detachedConfiguration85'.

[2020-03-11T19:24:55.748Z]    > Could not resolve org.nodejs:node:12.16.1.

[2020-03-11T19:24:55.748Z]      Required by:

[2020-03-11T19:24:55.748Z]          project :one-ui

[2020-03-11T19:24:55.748Z]       > Could not resolve org.nodejs:node:12.16.1.

[2020-03-11T19:24:55.748Z]          > Could not get resource 'https://nodejs.org/dist/v12.16.1/ivy.xml'.

[2020-03-11T19:24:55.748Z]             > Could not GET 'https://nodejs.org/dist/v12.16.1/ivy.xml'.

[2020-03-11T19:24:55.748Z]                > peer not authenticated

[2020-03-11T19:24:55.748Z] 

[2020-03-11T19:24:55.748Z] * Try:

[2020-03-11T19:24:55.748Z] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

[2020-03-11T19:24:55.748Z] 

[2020-03-11T19:24:55.748Z] * Get more help at https://help.gradle.org
```

**Comment by wooldridge** [comment]
@SameeraPriyathamTadikonda, this might be of help:

https://github.com/srs/gradle-node-plugin/issues/301

**Comment by SameeraPriyathamTadikonda** [comment]
> @SameeraPriyathamTadikonda, this might be of help:
> 
> [srs/gradle-node-plugin#301](https://github.com/srs/gradle-node-plugin/issues/301)

@wooldridge  Submitted a PR to fix the issue:
https://github.com/marklogic/marklogic-data-hub/pull/3689

**Comment by wooldridge** [comment]
@ryanjdew, looks like unit tests for the CSV functionality are failing according to Jenkins.

**Comment by SameeraPriyathamTadikonda** [comment]
Please rebase and resolve the conflicts.

**Pull request merged**